UnitOfWork/Services/IRepositories/IGenericRepository.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace UnitOfWork.Services.IRepositories;
+
+public interface IGenericRepository<T> where T : class
+{
+    Task<IEnumerable<T>> GetAll();
+    Task<T> GetById(int id);
+    Task<bool> Add(T entity);
```

Comment:
It's better to return a newly created entity `Task<T> Add(T entity);`

Reply:
fixed